WordPress/WordPressTest/KeychainTests.swift

```diff
@@ -0,0 +1,168 @@
+import XCTest
+@testable import WordPress
+
+class KeychainTests: XCTestCase {
+
+    let username = "Username"
+    let password = "Password"
+    let service = "Service"
+    let sharedGroup = WPAppKeychainAccessGroup
+
+    override func setUp() {
+        super.setUp()
+
+        SFHFKeychainUtilsMock.configure(with: [:])
+    }
+
+    func testNilAppGroupSavesToSharedGroupWhenFeatureEnabled() {
+        let subject = Keychain(useSharedKeychain: true, keychainUtils: SFHFKeychainUtilsMock.self)
+
+        try? subject.storeUsername(username, password: password, serviceName: service, updateExisting: true)
+
+        let result = getPassword(username: username, serviceName: service, accessGroup: sharedGroup)
+        XCTAssertEqual(result, password)
+    }
+
+    func testNilAppGroupDoesNotSaveToSharedGroupWhenFeatureDisabled() {
+        let subject = Keychain(useSharedKeychain: false, keychainUtils: SFHFKeychainUtilsMock.self)
+
+        try? subject.storeUsername(username, password: password, serviceName: service, updateExisting: true)
+
+        let result = getPassword(username: username, serviceName: service, accessGroup: sharedGroup)
+        XCTAssertNotEqual(result, password)
+    }
+
+    func testFeatureFlagChanging() {
+        var enabled = false
+        let subject = Keychain(useSharedKeychain: enabled, keychainUtils: SFHFKeychainUtilsMock.self)
```

Comment:
We can just pass in `false` I think instead of setting this to false and then to true.

Reply:
This test is actually testing that it's responsive to the `enabled` flipping from `false` to `true`. If we just pass in `false` here, the test will fail.

I added this test to account for someone toggling the feature flag in the app menu.

Reply:
But we manually set `enabled = true` regardless of what that function does don't we?  If the function is flipping it, shouldn't we remove line 39?

Reply:
The function isn't flipping it. It just uses the bool and does something different if it's `true` or `false`.

Manually setting it to `true` is to simulate someone flipping the feature flag to `true`. If you run the test with `enabled = true` commented out, the test fails.